grails-app/views/order/_orderSummary.gsp

```diff
@@ -80,14 +60,13 @@
 
             <tbody>
 
-            <g:set var="orderItemsDerivedStatus" value="${orderInstance?.getOrderItemsDerivedStatus()}"/>
-            <g:each var="orderItem" in="${orderInstance?.orderItems?.sort { a,b -> a.dateCreated <=> b.dateCreated ?: a.orderIndex <=> b.orderIndex }}" status="i">
-                <g:if test="${!orderItem?.canceled || orderInstance?.isPurchaseOrder}">
-                    <tr class="order-item ${(i % 2) == 0 ? 'even' : 'odd'} dataRow" style="${isItemCanceled ? 'background-color: #ffcccb;' : ''}">
-                        <g:if test="${orderInstance?.isPurchaseOrder}">
+            <g:each var="orderItem" in="${orderItems?.sort { a,b -> a.dateCreated <=> b.dateCreated ?: a.orderIndex <=> b.orderIndex }}" status="i">
+                <g:if test="${!orderItem?.canceled || isPurchaseOrder}">
+                    <tr class="order-item ${(i % 2) == 0 ? 'even' : 'odd'} dataRow" style="${orderItem?.canceled ? 'background-color: #ffcccb;' : ''}">
+                        <g:if test="${isPurchaseOrder}">
                             <td>
                                 <div class="tag ${orderItem?.canceled ? 'tag-danger' : ''}">
-                                    <format:metadata obj="${!orderItem?.canceled && orderItemsDerivedStatus[orderItem?.id] ? orderItemsDerivedStatus[orderItem?.id] : orderItem?.orderItemStatusCode?.name()}"/>
+                                    <span class="${orderItem?.id}">${g.message(code: 'default.loading.label')}</span>
```

Comment:
This feels odd. Why not use id=${orderItem.id} and class could be orderItemStatus or something like that.

Reply:
@jmiranda It's done on purpose because the derived status is used on the first and second tabs and item id is used to find elements and replace them. If I'd use an id property here instead of class, then after entering the second tab only ids on the first tab (hidden at the moment) would be replaced with statuses and the second tab would be hanging with "loading..." labels, because jquery would replace the first id it found and would be happy about it.